### PR TITLE
Remove globalopt dependency on the name of the entrypoint

### PR DESCRIPTION
**Pull request by MrTrillian**
GlobalOpt has an explicit test for a function called "main" (eww...) to enable an optimization. This updates the test to match the entry point, independent of its name.

Fixes #2350

### CODE REVIEW

File: tools/clang/test/CodeGenHLSL/batch/passes/globalopt/entrypoint_name_invariant_main.hlsl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// RUN: %dxc -T vs_6_0 -E main %s | FileCheck %s
```
Comment (MrTrillian) [inline tools/clang/test/CodeGenHLSL/batch/passes/globalopt/entrypoint_name_invariant_main.hlsl]: If I get a chance, I'll implement support for multiple RUN lines, this is a perfect use case.

**Comment by vcsharma** [comment]
It would be good to do it other way round. That is, enable this optimization for HLSL entry point when its name is not "main".  You may also want to add a test for hull shader which has additional patch constant function.

**Comment by vcsharma** [comment]
Since you have enabled back opt for HLSL entrypoints, i think your PR description needs an update..

**Comment by MrTrillian** [comment]
@vcsharma The bitcasts were coming from the `asuint`/`asfloat` in the original snippet, they are not a result of the bug. I've added a test for the patch constant function. I prefer keeping the helper where it is because I'm now preserving the fallback for testing for "main" from the original code

**Pull request merged**